File: dotnet/Multiplayer/Matchmaking/MatchmakingListResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Matchmaking
{
    /// <summary>
    /// Response containing a list of available matchmaking lobbies.
    /// </summary>
    /// <typeparam name="T">The type to deserialize lobby rules into.</typeparam>
    public class MatchmakingListResponse<T> : ApiResponse<EMatchmakingListError> where T : class, new()
    {
        /// <summary>
        /// List of available matchmaking lobbies with their details.
        /// </summary>
        public List<MatchmakingLobby<T>> Lobbies { get; set; } = new();
    }
}

File: java/src/main/java/com/michitai/multiplayer/matchmaking/MatchmakingListResponse.java
package com.michitai.multiplayer.matchmaking;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

import java.util.List;

/**
 * Response containing a list of available matchmaking lobbies.
 *
 * @param <T> The type to deserialize lobby rules into.
 */
public class MatchmakingListResponse<T> extends ApiResponse {
    @JsonProperty("lobbies")
    private List<MatchmakingLobby<T>> lobbies;

    @JsonProperty("count")
    private int count;

    public List<MatchmakingLobby<T>> getLobbies() {
        return lobbies;
    }

    public void setLobbies(List<MatchmakingLobby<T>> lobbies) {
        this.lobbies = lobbies;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }
}
